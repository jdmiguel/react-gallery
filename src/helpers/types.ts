export type ThumbData = {
  id: number;
  src: string;
  title: string;
}

export type DetailData = {
  id: number;
  src: string;
  title: string;
  views: number;
  downloads: number;
  favorites: number;
  likes: number;
}

export type ImageExtendedData = {
  id: number;
  pageURL: string;
  type: string;
  tags: string;
  previewURL: string;
  previewWidth: number;
  previewHeight: number;
  webformatURL: string;
  webformatWidth: number;
  webformatHeight: number;
  largeImageURL: string;
  fullHDURL: string;
  imageURL: string;
  imageWidth: number;
  imageHeight: number;
  imageSize: number;
  views: number;
  downloads: number;
  favorites: number;
  likes: number;
  comments: number;
  user_id: number;
  user: string;
  userImageURL: string;
}