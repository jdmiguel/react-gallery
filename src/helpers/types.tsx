export type Image = {
  src: string;
}