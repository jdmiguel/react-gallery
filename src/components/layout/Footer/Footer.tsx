import React from 'react';
import styled from 'styled-components';

const StyledFooter = styled.footer`
  position: absolute;
  width: 100%;
  left: 0;
  bottom: 0;
  text-align: center;
  padding: 25px 0;
  background-color: ${(props) => props.theme.palette.SECONDARY};
  display: flex;
  justify-content: center;
`;

const StyledText = styled.p`
  color: ${(props) => props.theme.palette.LIGHT_MEDIUM};
  font-size: 0.8rem;
  &:not(:first-of-type) {
    margin-left: 5px;
    @media only screen and (min-width: 375px) {
      margin-left: 7px;
    }
  }
  &:not(:last-of-type) {
    &:after {
      content: ' | ';
      margin-left: 3px;
      @media only screen and (min-width: 375px) {
        margin-left: 5px;
      }
    }
  }
  &:last-of-type {
    margin-right: 1px;
    @media only screen and (min-width: 375px) {
      margin-right: 3px;
    }
  }
  @media only screen and (min-width: 375px) {
    font-size: 0.9rem;
  }
  @media only screen and (min-width: 768px) {
    font-size: 1rem;
  }
`;

const StyledLink = styled.a`
  color: ${(props) => props.theme.palette.PRIMARY};
  font-weight: 500;
  font-size: 0.8rem;
  text-decoration: none;
  transition: color 0.3s ease-out;
  &:hover {
    color: ${(props) => props.theme.palette.PRIMARY_MEDIUM};
  }
  margin-left: 3px;
  @media only screen and (min-width: 375px) {
    margin-left: 5px;
    font-size: 0.9rem;
  }
  @media only screen and (min-width: 768px) {
    font-size: 1rem;
  }
`;

const Footer: React.FC = () => (
  <StyledFooter>
    <StyledText>REACT GALLERY</StyledText>
    <StyledText>
      REPO ON 
      <StyledLink href="https://github.com/jdmiguel/react-gallery" target="_blank">
        GITHUB
      </StyledLink>
    </StyledText>
    <StyledText>
      BY
      <StyledLink href="https://jdmiguel.netlify.app" target="_blank">
        JDMIGUEL
      </StyledLink>
    </StyledText>
  </StyledFooter>
);

export default Footer;
