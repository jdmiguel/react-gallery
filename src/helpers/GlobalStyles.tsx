import { createGlobalStyle } from 'styled-components';

export const GlobalStyles = createGlobalStyle`
  body {
    margin: 0;
    padding: 0;
    font-family: 'Josefin Sans', sans-serif;
    font-size: 15px;
    color: #2f3235;
    background-color: #F3F3F3;
  }

  h1, h2, h3, h4, h5, h6, p{
    margin: 0;
  }

  #root{
    overflow: auto;
  }
`;