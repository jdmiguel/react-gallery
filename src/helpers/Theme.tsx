import React, { ReactNode } from 'react';
import { DefaultTheme, ThemeProvider } from 'styled-components';

export const theme: DefaultTheme = {
  palette: {
    PRIMARY: '#A66885',
    PRIMARY_DARK: '#734359',
    PRIMARY_LIGHT: '#F2C1AE',
    LIGHT_MAX: '#E5EAED',
    LIGHT_MIN: '#C8C8C8',
    GREY: '#E5EAED',
    DARK_MAX: '#2f3235',
    DARK_MIN: '#798186',
  },
};

export const renderWithTheme = (component: ReactNode) => (
  <ThemeProvider theme={theme}>{component}</ThemeProvider>
);
