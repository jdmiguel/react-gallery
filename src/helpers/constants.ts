// API
export const API_PIXABAY = "https://pixabay.com/api/";
export const API_KEY = "6473511-0417f2cad683f1bee54cafe15";

// IMAGES
export const DEFAULT_THUMBS = [
  {
    id: 0,
    src: '',
    title: ''
  },
];