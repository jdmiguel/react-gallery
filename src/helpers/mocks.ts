export const THUMB_PROPS_MOCKED = {
  id: 3063284,
  src: "https://pixabay.com/get/55e0d340485aa814f1dc84609629327f113dd7e05b4c704f752f78d79744c15e_640.jpg",
  title: "flower"
};
export const BUTTON_TEXT_MOCKED = "default button";
export const Pill_TEXT_MOCKED = "default pill";