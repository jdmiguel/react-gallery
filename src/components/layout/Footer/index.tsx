import React from 'react';
import styled from 'styled-components';
import { InView } from 'react-intersection-observer';

import { OBSERVER_OPTIONS } from '../../../helpers/constants';

export type FooterProps = {
  onViewFooter: () => void;
}

const StyledFooter = styled.footer`
  position: absolute;
  width: 100%;
  left: 0;
  bottom: 0;
  text-align: center;
  padding: 25px 0;
  background-color: ${({theme}) => theme.palette.PRIMARY_DARK};
  display: flex;
  justify-content: center;
`;

const StyledText = styled.p`
  color: ${({theme}) => theme.palette.PRIMARY_LIGHT};
  font-size: 0.7rem;
  &:not(:first-of-type) {
    margin-left: 5px;
    @media only screen and (min-width: 375px) {
      margin-left: 7px;
    }
  }
  &:not(:last-of-type) {
    &:after {
      content: ' | ';
      margin-left: 3px;
      @media only screen and (min-width: 375px) {
        margin-left: 5px;
      }
    }
  }
  &:last-of-type {
    margin-right: 1px;
    @media only screen and (min-width: 375px) {
      margin-right: 3px;
    }
  }
  @media only screen and (min-width: 375px) {
    font-size: 0.8rem;
  }
  @media only screen and (min-width: 768px) {
    font-size: 1rem;
  }
`;

const StyledLink = styled.a`
  color: ${(props) => props.theme.palette.LIGHT_MAX};
  font-weight: 500;
  font-size: 0.7rem;
  text-decoration: none;
  transition: color 0.3s ease-out;
  &:hover {
    color: ${(props) => props.theme.palette.PRIMARY_MEDIUM};
  }
  margin-left: 3px;
  @media only screen and (min-width: 375px) {
    margin-left: 5px;
    font-size: 0.8rem;
  }
  @media only screen and (min-width: 768px) {
    font-size: 1rem;
  }
`;

const Footer: React.FC<FooterProps> = ({ onViewFooter }) => (
  <InView
    {...OBSERVER_OPTIONS}
    as="div"
    onChange={(inView) => {
      if (inView) {
        onViewFooter();
      }
    }}
  >
    <StyledFooter>
      <StyledText>REACT GALLERY</StyledText>
      <StyledText>
        REPO ON 
        <StyledLink href="https://github.com/jdmiguel/react-gallery" target="_blank">
          GITHUB
        </StyledLink>
      </StyledText>
      <StyledText>
        BY
        <StyledLink href="https://jdmiguel.netlify.app" target="_blank">
          JDMIGUEL
        </StyledLink>
      </StyledText>
    </StyledFooter>
  </InView>
);

export default Footer;
