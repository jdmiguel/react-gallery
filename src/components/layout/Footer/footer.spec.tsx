import React from 'react';
import { render } from '@testing-library/react';

import Footer from '.';

import { renderWithTheme } from '../../../helpers/Theme';

describe('Component: Footer', () => {
  it('should render', () => {
    const { container } = render(
      renderWithTheme(<Footer />)
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});