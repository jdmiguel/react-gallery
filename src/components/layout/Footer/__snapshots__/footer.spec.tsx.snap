// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Footer should render 1`] = `
.c0 {
  position: absolute;
  width: 100%;
  left: 0;
  bottom: 0;
  text-align: center;
  padding: 25px 0;
  background-color: #734359;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  color: #F2C1AE;
  font-size: 0.7rem;
}

.c1:not(:first-of-type) {
  margin-left: 5px;
}

.c1:not(:last-of-type):after {
  content: ' | ';
  margin-left: 3px;
}

.c1:last-of-type {
  margin-right: 1px;
}

.c2 {
  color: #E5EAED;
  font-weight: 500;
  font-size: 0.7rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: color 0.3s ease-out;
  transition: color 0.3s ease-out;
  margin-left: 3px;
}

@media only screen and (min-width:375px) {
  .c1:not(:first-of-type) {
    margin-left: 7px;
  }
}

@media only screen and (min-width:375px) {
  .c1:not(:last-of-type):after {
    margin-left: 5px;
  }
}

@media only screen and (min-width:375px) {
  .c1:last-of-type {
    margin-right: 3px;
  }
}

@media only screen and (min-width:375px) {
  .c1 {
    font-size: 0.8rem;
  }
}

@media only screen and (min-width:768px) {
  .c1 {
    font-size: 1rem;
  }
}

@media only screen and (min-width:375px) {
  .c2 {
    margin-left: 5px;
    font-size: 0.8rem;
  }
}

@media only screen and (min-width:768px) {
  .c2 {
    font-size: 1rem;
  }
}

<footer
  class="c0"
>
  <p
    class="c1"
  >
    REACT GALLERY
  </p>
  <p
    class="c1"
  >
    REPO ON
    <a
      class="c2"
      href="https://github.com/jdmiguel/react-gallery"
      target="_blank"
    >
      GITHUB
    </a>
  </p>
  <p
    class="c1"
  >
    BY
    <a
      class="c2"
      href="https://jdmiguel.netlify.app"
      target="_blank"
    >
      JDMIGUEL
    </a>
  </p>
</footer>
`;
